File: atividade-23/atividade.go
package main

import (
	"fmt"
	"regexp"
)

type AFN struct{}

func (AFN) IsAccepted(string string) bool {
	pattern := regexp.MustCompile("101|110") 
	return pattern.MatchString(string)       
}

func main() {
	afn := AFN{}
	fmt.Println(afn.IsAccepted("101"))  
	fmt.Println(afn.IsAccepted("110"))  
	fmt.Println(afn.IsAccepted("0101")) 
	fmt.Println(afn.IsAccepted("0110")) 
	fmt.Println(afn.IsAccepted("1110")) 
	fmt.Println(afn.IsAccepted("000"))  
	fmt.Println(afn.IsAccepted("001"))  
}

File: atividade-23/atividade.py
import re

class AFN:
    def is_accepted(self, string):
        padrao = r"101|110"
        return bool(re.search(padrao, string))

if __name__ == '__main__':
    afn = AFN()
    print(afn.is_accepted("101"))    # True
    print(afn.is_accepted("110"))    # True
    print(afn.is_accepted("0101"))   # True
    print(afn.is_accepted("0110"))   # True
    print(afn.is_accepted("1110"))   # True
    print(afn.is_accepted("000"))    # False
    print(afn.is_accepted("001"))    # False